chat-plugins/datasearch.js

```diff
@@ -1377,29 +1378,14 @@ function runItemsearch(target, cmd, canAll, message) {
 				if (descWords.includes(word)) matched++;
 			}
 
-			if (matched >= bestMatched && matched >= (searchedWords.length * 3 / 5)) foundItems.push(item.name);
+			if (matched >= bestMatched && matched >= (searchedWords.length * 3 / 5)) {
```

Comment:
Honestly, you don't even need the `matchCountMap`. Just do `if (matched > bestMatched) foundItems = [];` here.

Reply:
And then you don't need to do the filter at all.

Reply:
Wouldn't that require all matches to be sorted by their ``matched`` first?

Reply:
No, `if (matched > bestMatched) foundItems = [];` does the same thing.

Like, the goal here is to get a list of results tied for maximum `matched`, right?

In which case, that line is all you need. Just think through the inductive hypothesis.

- New thing < bestMatched – ignore

- New thing = bestMatched – add to list of best matches

- New thing > bestMatched – empty the previous list of best matches, add to new empty list